### ISSUE TITLE
[bug] Wrong comments count for an updated research item

### ISSUE THREAD

**Issue by codisart** [ONEARMY/community-platform#3510]
**Describe the bug**

When looking at the comments posted by @benfurber trying to reproduce the bug for #3508, I think I found a bug with the count of comments for a research

**To Reproduce**
Steps to reproduce the behaviour:

1. Go to `/research/lc-test-240506-0059`
2. See on the research statistics that the count is 6
3. Scroll down to each update
4. See that the counts is respectely 2 and 6

**Expected behaviour**

The count should be 8

**Screenshots**

<img width="667" alt="Screenshot 2024-05-08 at 21 15 09" src="https://github.com/ONEARMY/community-platform/assets/1767237/f0d2eac4-f960-4eed-8b88-1fed51b2a70a">

**Additional context**

I think the bug come from possible race conditions with 

```typescript
await dbRef.update({
  mentions: users
    .map((userName) => ({
      username: userName,
      location: `update-${existingUpdateIndex}-comment:${newComment?._id}`,
    }))
    .concat(researchItem.mentions || []),
  totalCommentCount: (researchItem.totalCommentCount || 0) + 1,
} as any)
```

**Comment by codisart** [comment]
Hi @benfurber, did you delete your two comments from the update 1 of `/research/lc-test-240506-0059` ? It seems that I managed to add a reply after that it is considered valid and counted but can not be displayed. Does it seem possible to you ?

**Comment by benfurber** [comment]
> Hi @benfurber, did you delete your two comments from the update 1 of `/research/lc-test-240506-0059` ? It seems that I managed to add a reply after that it is considered valid and counted but can not be displayed. Does it seem possible to you ?

Yes, another bug you've found! Could you write up in another issue how you think that could be implemented please?

**Comment by mariojsnunes** [comment]
I found two issues:
1. There are two variables tracking the count: `totalCommentCount` and `commentCount`, should be only 1.
Currently the search is using `totalCommentCount` for sorting so would be easier to use that.
2. Where are we adding/subtracting the research update comments to the research comment count?

**Comment by benfurber** [comment]
@mariojsnunes Sorry, I should have assigned this to myself. I've got a fix in #3450 and then was going to refactor all the unneccessarily different commentCount naming in a follow-up.

**Comment by dalibormrska** [comment]
Hey again! The issue with the comment counts seems to be on again.

Here you see the count being on 28, while the real amount is 12.
https://dev.onearmy.world/research/can-we-support-nested-comments
Maybe it is connected to me testing locally and experiencing seeing my comments being posted twice. But I have no idea.

I experienced other strange comment count problem, i don't know if related. On the project kamp instance, the ruin research post on mobile has NaN comments, on desktop it has 0 and in reality it has 2.
![telegram-cloud-photo-size-4-5985369310656316463-y](https://github.com/ONEARMY/community-platform/assets/35503298/687c8272-db13-4f96-a32b-f8656d194298)
<img width="1250" alt="image" src="https://github.com/ONEARMY/community-platform/assets/35503298/ae1123ec-2363-4dbd-a028-3b3313297da7">
